src/audio/volume/volume_hifi3.c

```diff
@@ -133,11 +159,17 @@ static void vol_s32_to_s24_s32(struct comp_dev *dev, struct audio_stream *sink,
 			mult = AE_MULF32S_LL(volume, in_sample);
 
 			/* Multiplication of Q1.31 x Q1.31 gives Q1.63.
-			 * Now multiplication is Q8.16 x Q1.31, the result
-			 * is Q9.48. Need to shift right by one to get Q17.47
-			 * compatible format for round.
+			 * Now multiplication is Q8.16 x Q1.31 (IPC3) or Q1.23 x
+			 * Q1.31 (IPC4), the result is appropriately
+			 * Q9.48 (IPC3) or Q1.55 (IPC4). Need to shift right by
+			 * 1 (IPC3) or 8 (IPC4) to get Q17.47 compatible format
+			 * for round.
 			 */
+#if CONFIG_IPC_MAJOR_3
```

Comment:
Maybe it would be better to have separate kconfig macro for Q1.23 and Q8.16 volume modes and select defaults with IPC version. I'm not really happy with Q8.16. We did it because of least incompatibility issues when the Q1.16 gain range was extended. Not a blocker but I think more logical macros naming would be nicer for long run.